docs/source/notes/cuda.rst

```diff
@@ -54,6 +54,68 @@ Below you can find a small example showcasing this::
         f = torch.randn(2).cuda(cuda2)
         # d.device, e.device, and f.device are all device(type='cuda', index=2)
 
+.. _tf32_on_ampere:
+
+TensorFloat-32(TF32) on Ampere devices
+--------------------------------------
+
+Starting in PyTorch 1.7, there is a new flag called `allow_tf32` which defaults to true.
+This flag controls whether PyTorch is allowed to use the TensorFloat32 (TF32) tensor cores,
+available on new NVIDIA GPUs since Ampere, internally to compute GEMMs and convolutions.
+
+TF32 tensor cores are designed to achieve better performance on GEMMs and convolutions on
+`torch.float32` tensors by doing arithmetics at the precision of `torch.half` while still
+maintaining the dynamic range of `torch.float32`.
+
+GEMMs and convolutions are controlled separately, and their corresponding flag can be accessed at:
+
+.. code:: python
+
+  # The flag below controls whether to allow TF32 on GEMMs. This flag defaults to True.
+  torch.backends.cuda.matmul.allow_tf32 = True
+
+  # The allow_tf32 flag for convolutions is not implemented yet
+
+To get an idea of the precision and speed, see the example code below:
+
+.. code:: python
+
+  a_full = torch.randn(10240, 10240, dtype=torch.double, device='cuda')
+  b_full = torch.randn(10240, 10240, dtype=torch.double, device='cuda')
+  ab_full = a_full @ b_full
+
+  a = a_full.float()
+  b = b_full.float()
+
+  # Do matmul at TF32 mode.
+  ab_tf32 = a @ b  # takes 0.018s on GA100
+  error = (ab_tf32 - ab_full).abs().max()  # 0.1713
+
+  # Do matmul with TF32 disabled.
+  torch.backends.cuda.matmul.allow_tf32 = False
+  ab_fp32 = a @ b  # takes 0.14s on GA100
+  error = (ab_fp32 - ab_full).abs().max()  # 0.0028
+  
+From the above example, we can see that with TF32 enabled, the speed is ~8x faster, but the precision
+has ~2 digits less in the result.  For most AI applications, the default value `allow_tf32 = True` is
```

Comment:
This documentation is very good and greatly improved over the original paragraph, but it can't be so confident that it asserts `allow_tf32=True` is the "optimal choice... for most AI applications." 

You may just want to say, "if the additional precision is needed..."

Reply:
fixed